src/petals/utils/ping.py

```diff
@@ -0,0 +1,60 @@
+import asyncio
+import math
+import time
+from functools import partial
+from typing import Dict, Sequence
+
+import hivemind
+from hivemind.proto import dht_pb2
+from hivemind.utils.logging import get_logger
+
+logger = get_logger(__name__)
+
```

Comment:
This file is a rework of https://github.com/petals-infra/health.petals.ml/blob/main/p2p_utils.py